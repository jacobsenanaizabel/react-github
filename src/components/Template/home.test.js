import React from 'react'
import { shallow, mount } from 'enzyme'
import '../../setupTests'

import Home from './home'

describe('<Home />', () => { 
    it('renders home', () => {
        shallow(<Home />);
    })

    it('render titles in home', () => {
        let wrapper = shallow(<Home />);
    
        expect(wrapper.contains(<div className="title welcome" />));
        expect(wrapper.text()).toContain('Ache Seu Desenvolvedor Favorito');

        expect(wrapper.contains(<div className="subtitle" />));
        expect(wrapper.text()).toContain('De maneira muito rapida e facil, apenas digite');
    })

    it('simulate search user event', () =>{
        let wrapper = shallow(<Home />);

        expect(wrapper.find('button').simulate('click'));
    })
});