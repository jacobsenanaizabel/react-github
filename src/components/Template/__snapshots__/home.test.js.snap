// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> renders home snapshot 1`] = `
<MemoryRouter
  keyLength={0}
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Home>
      <div
        className="home-page row"
      >
        <div
          className="banner-home col-md-8"
        >
          <h1
            className="title welcome"
          >
            Ache Seu Desenvolvedor Favorito
          </h1>
          <h2
            className="subtitle"
          >
            De maneira muito rapida e facil, apenas digite :
          </h2>
          <div
            className="col-md-6"
          >
            <div
              className="input-group"
            >
              <input
                className="form-control search-user"
                name="email"
                onChange={[Function]}
                placeholder="Procure por..."
                type="text"
                value=""
              />
              <span
                className="input-group-btn"
              >
                <button
                  className="btn btn-default"
                  type="button"
                >
                  <Link
                    replace={false}
                    to={
                      Object {
                        "pathname": "/search",
                        "state": Object {
                          "user": "",
                        },
                      }
                    }
                  >
                    <a
                      href="/search"
                      onClick={[Function]}
                    >
                       Vai 
                    </a>
                  </Link>
                </button>
              </span>
            </div>
          </div>
        </div>
        <Component>
          <footer
            class="footer"
          >
            <div
              class="container"
            >
              <span>
                Feito com ‚ù§ por Ana Jacobsen.
              </span>
            </div>
          </footer>
        </Component>
      </div>
    </Home>
  </Router>
</MemoryRouter>
`;
